x-pack/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -70,7 +70,9 @@ export function App({
     const currentRange = data.query.timefilter.timefilter.getTime();
     return {
       query: data.query.queryString.getQuery(),
-      filters: data.query.filterManager.getFilters(),
+      filters: !initialContext
+        ? data.query.filterManager.getGlobalFilters()
+        : data.query.filterManager.getFilters(),
```

Comment:
Stylistic comment: could you flip the branches of this ternary? This might simplify the understanding of the code.

Also since we've had issues with this part of the code, maybe it deserves a code comment here?